src/FEValues/common_values.jl

```diff
@@ -2,14 +2,23 @@
 
 using Base: @propagate_inbounds
 
-const ScalarValues{dim,T,shape} = Union{CellScalarValues{dim,T,shape},FaceScalarValues{dim,T,shape},PointScalarValues{dim,T,shape}}
-const VectorValues{dim,T,shape} = Union{CellVectorValues{dim,T,shape},FaceVectorValues{dim,T,shape}}
+abstract type FieldTrait end
+struct IsVectorValued <: FieldTrait end
+struct IsScalarValued <: FieldTrait end
+
+FieldTrait(::Type{<:CellScalarValues}) = IsScalarValued()
+FieldTrait(::Type{<:FaceScalarValues}) = IsScalarValued()
+FieldTrait(::Type{<:PointScalarValues}) = IsScalarValued()
+FieldTrait(::Type{<:CellVectorValues}) = IsVectorValued()
+FieldTrait(::Type{<:FaceVectorValues}) = IsVectorValued()
```

Comment:
I really like the idea. 

Maybe we should be a bit more precise here and adjust the name? I mean, we could utilize the FieldTraits for more things like e.g. the conformity-class (e.g. H1, L2, H(div), H(curl), ...).

My first suggestion would be `FieldValueTrait` - however I also do not like this name too much.

Reply:
To give one example, H(curl) and H(div) need different transformations from and to the reference element than L2 and H1 conforming formulations. See e.g. https://www.cambridge.org/core/services/aop-cambridge-core/content/view/4B91C5FDC5131BE591D41F8577DEB4F3/S0764583X10000348a.pdf/hexahedral-hdiv-and-hcurl-finite-elements-.pdf .

However, this was primarily ment to give an example why the name "field trait" may be ambiguous. :)

Reply:
Wouldn't that be handled by having special `reinit!` functions for e.g. some `HcurlValues`? 

I'm open to a better name suggestion though, no hard opinions there. `FieldTrait` would be internal to Ferrite anyways, so technically non-breaking to change (even later) it if there ever would be a hard need for a different name.

Reply:
Depends on how we want to design it. Technically speaking a CellVectorValues could do the job here already, right @koehlerson ?

Reply:
If the `reinit!` would dispatch on the the trait: yes. Otherwise we would need a new cellvalue type

Reply:
on the other hand one could just dispatch on the cellvalue type, mhm don't know :man_shrugging:

Reply:
My cohesive zone elements are why I want to do this at all.

I have a `SurfaceVectorValues` type that has slightly different fields and a custom `reinit!`, but I want to use all the other stuff which is defined for `VectorValues`. But before this PR, there is no way to tell Ferrite that `SurfaceVectorValues` are `VectorValues`.

Not sure if using a trait for the `reinit!` function would be the right design. As it is now this PR is just meant to allow extending the `VectorValues` and `ScalarValues`, but doesn't change anything about the programming logic of them. Perhaps we should open as separate issue with a concrete example for you other elements in order to discuss general design stuff?

Reply:
Okay sorry I was confused, I totally see the point now. I think it's worth to discuss this further in an issue :)